src/main/java/de/tum/in/www1/artemis/web/rest/FileUploadExerciseResource.java

```diff
@@ -142,41 +139,38 @@ private void validateNewOrUpdatedFileUploadExercise(FileUploadExercise fileUploa
 
         if (!isFilePatternValid(fileUploadExercise)) {
             throw new BadRequestAlertException("The file pattern is invalid. Please use a comma separated list with actual file endings without dots (e.g. 'png, pdf').",
-                    ENTITY_NAME, "filepattern.invalid");
+                    ENTITY_NAME, "filePattern.invalid");
         }
     }
 
     /**
-     * PUT /file-upload-exercises : Updates an existing fileUploadExercise.
+     * PUT file-upload-exercises : Updates an existing FileUploadExercise.
      *
-     * @param fileUploadExercise the fileUploadExercise to update
+     * @param fileUploadExercise the FileUploadExercise to update
      * @param notificationText the text shown to students
      * @param exerciseId the id of exercise
      * @return the ResponseEntity with status 200 (OK) and with body the updated fileUploadExercise, or with status 400 (Bad Request) if the fileUploadExercise is not valid, or
      *         with status 500 (Internal Server Error) if the fileUploadExercise couldn't be updated
      */
-    @PutMapping("/file-upload-exercises/{exerciseId}")
+    @PutMapping("file-upload-exercises/{exerciseId}")
     @PreAuthorize("hasRole('EDITOR')")
     public ResponseEntity<FileUploadExercise> updateFileUploadExercise(@RequestBody FileUploadExercise fileUploadExercise,
             @RequestParam(value = "notificationText", required = false) String notificationText, @PathVariable Long exerciseId) {
         log.debug("REST request to update FileUploadExercise : {}", fileUploadExercise);
 
-        // TODO: The route has an exerciseId but we don't do anything useful with it. Change route and client requests?
+        if (!fileUploadExercise.getId().equals(exerciseId)) {
+            return badRequest("FileUploadExerciseID", "400", "exerciseId in path doesn't match the exerciseId in the Body!");
+        }
 
         // Validate the updated file upload exercise
         validateNewOrUpdatedFileUploadExercise(fileUploadExercise);
         // validates general settings: points, dates
         exerciseService.validateGeneralSettings(fileUploadExercise);
 
-        // Retrieve the course over the exerciseGroup or the given courseId
-        Course course = courseService.retrieveCourseOverExerciseGroupOrCourseId(fileUploadExercise);
-
         // Check that the user is authorized to update the exercise
-        User user = userRepository.getUserWithGroupsAndAuthorities();
-        if (!authCheckService.isAtLeastEditorInCourse(course, user)) {
-            return forbidden();
-        }
         FileUploadExercise fileUploadExerciseBeforeUpdate = fileUploadExerciseRepository.findOneByIdElseThrow(fileUploadExercise.getId());
+        User user = userRepository.getUserWithGroupsAndAuthorities();
+        authCheckService.checkHasAtLeastRoleForExerciseElseThrow(Role.EDITOR, fileUploadExerciseBeforeUpdate, user);
```

Comment:
```suggestion
        authCheckService.checkHasAtLeastRoleForExerciseElseThrow(Role.EDITOR, fileUploadExerciseBeforeUpdate, null);
```

Reply:
Here, the user is needed a little further down the line for log messages, so might as well fetch it here ^^